cmake/modules/AddSwift.cmake

```diff
@@ -488,6 +488,20 @@ function(_add_variant_link_flags)
     endif()
   endif()
 
+  # Enable dead stripping. Portions of this logic were copied from llvm's
+  # `add_link_opts` function (which, perhaps, should have been used here in the
+  # first place, but at this point it's hard to say whether that's feasible).
+  #
+  # TODO: Evaluate/enable -f{function,data}-sections --gc-sections for bfd,
```

Comment:
BFD is an unsupported linker (it cannot handle the protected semantics needed for swift)